### ISSUE TITLE
chapter 9 "Star/Quasar Classification ROC Curves" example trains classifiers on the whole data set rather than the train split

### ISSUE THREAD

**Issue by StKyr** [astroML/astroML_figures#50]
In *fig_star_quasar_ROC.py*, inside `compute_results()`, classifiers are trained on `X` rather than `X_train`. This means that the test set has been observed from the classifiers during training which of course is a bad practice.

The way to fix this would be to change line 90 from
```
model.fit(X, y)
```
to
```
model.fit(X_train, y_train)
```
Additionally, the figure *fig_star_quasar_ROC_1.png* needs to be updated as it is the result of the execution of the script.

**Comment by StKyr** [comment]
Created the relevant pull request, so I am closing the issue
